lib/creds.py

```diff
@@ -14,19 +14,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import os
+import subprocess
 
 
 def get_creds(args={}):
     """get credentials -- Note this does not currently push to
     myproxy, nor does it yet deal with tokens, but those should
     be done here as needed.
     """
-    f = os.popen("fake_ifdh getProxy","r") 
-    p = f.read().strip()
-    f.close()
-    f = os.popen("fake_ifdh getToken","r") 
-    t = f.read().strip()
-    f.close()
+    p = subprocess.run('fake_ifdh getProxy', shell=True, bufsize=0, stdout=subprocess.PIPE, universal_newlines=True).stdout   
+    t = subprocess.run('fake_ifdh getToken', shell=True, bufsize=0, stdout=subprocess.PIPE, universal_newlines=True).stdout
+
```

Comment:
Is there any reason shell=True here?  Similarly, is there a reason `stdout=subprocess.PIPE` and you're not using `capture_output=True`?  Not saying I don't like it, I'm just curious as to the reason it's done this way.

Also, do we want to check the output (setting `check=True`) and raise an error (`CalledProcessError` is what the `subprocess` module will raise) if either credential call fails?

Reply:
Error handling done

Reply:
Thanks for looking into that!